lib/src/clojure_lsp/snapshot.clj

```diff
@@ -0,0 +1,57 @@
+(ns clojure-lsp.snapshot
+  (:require
+   [clojure.java.io :as io]
+   [clojure.string :as str]))
+
+(defn ^:private parse [line]
+  (let [[_ file row column level linter message]
+        (re-find #"(.*):(\d+):(\d+): ([^:]+): \[([^\]]+)+] (.*)" line)
+        [source code] (str/split linter #"/")]
+    {:file file
+     :row (dec (Integer. row))
+     :column (dec (Integer. column))
+     :level level
+     :severity (case level
+                 "error" 1
+                 "warning" 2
+                 "info" 3)
+     :source source
+     :code code
+     :linter linter
+     :message message
+     :raw line}))
+
+(defn read-db
+  ([] (read-db ".lsp/snapshot.txt"))
```

Comment:
Just a reminder to handle this in a way work for windows and other places like `(Io/file ".lsp"  "snapshot.txt)`